ext/lmdb-safe/lmdb-safe.hh

```diff
@@ -99,54 +166,97 @@ struct MDBOutVal
     return d_mdbval;
   }
 
+#ifndef DNSDIST
   template <class T,
-          typename std::enable_if<std::is_arithmetic<T>::value,
+          typename std::enable_if<std::is_integral<T>::value,
                                   T>::type* = nullptr> const
   T get()
+  {
+    T ret;
+
+    size_t offset = LMDBLS::LScheckHeaderAndGetSize(this, sizeof(T));
+
+    memcpy(&ret, (char *)d_mdbval.mv_data+offset, sizeof(T));
+    ret = ntohl(ret);
+    return ret;
+  }
+
+  template <class T,
+          typename std::enable_if<std::is_arithmetic<T>::value,
+                                  T>::type* = nullptr> const
+  T getNoStripHeader()
   {
     T ret;
     if(d_mdbval.mv_size != sizeof(T))
       throw std::runtime_error("MDB data has wrong length for type");
 
     memcpy(&ret, d_mdbval.mv_data, sizeof(T));
+    ret = ntohl(ret);
     return ret;
   }
 
+#endif /* ifndef DNSDIST */
+
   template <class T,
             typename std::enable_if<std::is_class<T>::value,T>::type* = nullptr>
   T get() const;
 
+
+#ifndef DNSDIST
+  template <class T,
+            typename std::enable_if<std::is_class<T>::value,T>::type* = nullptr>
+  T getNoStripHeader() const;
+#endif
+
+#ifndef DNSDIST
   template<class T>
   T get_struct() const
   {
     T ret;
-    if(d_mdbval.mv_size != sizeof(T))
-      throw std::runtime_error("MDB data has wrong length for type");
 
-    memcpy(&ret, d_mdbval.mv_data, sizeof(T));
+    size_t offset = LMDBLS::LScheckHeaderAndGetSize(this, sizeof(T));
+
+    memcpy(&ret, (char *)d_mdbval.mv_data+offset, sizeof(T));
     return ret;
   }
 
   template<class T>
   const T* get_struct_ptr() const
   {
-    if(d_mdbval.mv_size != sizeof(T))
-      throw std::runtime_error("MDB data has wrong length for type");
+    size_t offset = LMDBLS::LScheckHeaderAndGetSize(this, sizeof(T));
 
-    return reinterpret_cast<const T*>(d_mdbval.mv_data);
+    return reinterpret_cast<const T*>((char *)d_mdbval.mv_data+offset);
```

Comment:
I'm worried about alignment here. Do we actually use this?

Reply:
we don't, and we don't even use the `get_struct` variant. (If we did, I'd be surprised at this point, because v5 tries to use network order everywhere so that databases become portable between systems).

I'll remove both methods.